chain/processor.go

```diff
@@ -80,8 +80,18 @@ func (b *StatelessBlock) Execute(
 			// Fetch keys from disk
 			var toCache map[string]*fetchData
 			if len(toLookup) > 0 {
+				cacheLock.RLock()
```

Comment:
I'm not sure a `RLock` helps prevent concurrent reads here as any number of transactions would see the value is missing in the cache and try to read it?